iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/variant.inl

```diff
@@ -31,7 +31,7 @@ inline constexpr variant<Types...>::variant(const variant& rhs) noexcept
 {
     if (m_type_index != INVALID_VARIANT_INDEX)
     {
-        internal::call_at_index<0, Types...>::copyConstructor(m_type_index, rhs.m_storage, m_storage);
+        internal::call_at_index<0, Types...>::copyConstructor(m_type_index, &rhs.m_storage.data[0], &m_storage.data[0]);
```

Comment:
We could probably get rid of the `.data` indirection by defining a conversion to a pointer to the start of the storage (as part of the storage). This is also only a suggestion if we decide to implement our own aligned_storage in another issue.

Generally the address of the struct will be the same as the address of data, so if a type erased pointer (e.g. `void*` or `char*`) is used this should not be needed.

Observation, not to be done in this issue.

Reply:
I removed `.data` now that I changed the `byte_t*` parameter in `call_at_index` to `void*` (before it was necessary).